### ISSUE TITLE
How to install Python 2.7?

### ISSUE THREAD

**Issue by gfixler** [msys2/MSYS2-packages#2334]
Python 2.7 has been removed from the list of packages, because it's support/life ended a year ago. However, those of us still in Autodesk Maya are stuck in 2.7 for the foreseeable future. I do all of my Maya Python dev work in MSYS2. How can I install it, despite this?

**Comment by 1480c1** [comment]
You can find some of the python2 packages here http://repo.msys2.org/mingw/x86_64/ and do 
```bash
pacman -U http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python2-2.7.18-1-any.pkg.tar.xz
```
although it would be recommended to download that file locally somewhere else since there is no guarantee that that URL will be valid anytime in the future

**Comment by gfixler** [comment]
That worked, but now that it installed from the local file, I can't find, nor launch Python.

**Comment by 1480c1** [comment]
Try checking under /mingw64/bin for python or python2

**Comment by Biswa96** [comment]
Though all python packages are built with python3. Why stick with old version when you can convert code to newer one?

**Comment by 1480c1** [comment]
I'm assuming he himself cannot convert the code as I'm assuming Autodesk Maya only supports 2.7.11 based on https://knowledge.autodesk.com/support/maya/learn-explore/caas/CloudHelp/cloudhelp/2020/ENU/Maya-Scripting/files/GUID-C0F27A50-3DD6-454C-A4D1-9E3C44B3C990-htm.html
```none
Version
Maya installs Python with your Maya installation. Maya uses Python version 2.7.11 on all supported platforms. The standalone Python shell for Maya is named mayapy.exe on Windows and mayapy on Linux and Mac OS X.
```

**Comment by gfixler** [comment]
> I'm assuming he himself cannot convert the code as I'm assuming Autodesk Maya only supports 2.7.11...

Correct. Many of us are desperate to switch to 3, as we're starting to run into problems like this. I've missed so many years of developments, and have a lot of catching up to do, some day. For now, even Maya 2021 is still on 2.7. The reason is the industries (film/games) are created now of 2.7 code, and switching to 3 would probably cause many studios to hold off updating, for fear of missing major release deadlines, having to rewrite whole, gigantic toolchains, which Autodesk doesn't want, so they stick with 2.7.

Yes, I found 2.7 in `/mingw64/bin`, but was getting a `LookupError: unknown encoding: cp65001` error. Long story short, my MSYS2's .bashrc
, which now has `export PYTHONIOENCODING=UTF-8`, which solved that problem, giving me 2.7 in MSYS2, thanks!

I've been running Maya tests (from Vim) through mayapy, and non-Maya tests through python2.7 in the shell for years (different hotkeys), because firing up the Maya libs in mayapy takes 6 seconds, and if they're not needed, I can get instant test results out in the shell. I don't know that it's ever occurred to me that I can just use mayapy without firing up the libs, and the wait, which seems a better idea, because then all the code is tested right in Maya's Python.

The problem then was that `/c/Program\ Files/Autodesk/Maya2018/bin/mayapy.exe` would sit there with no prompt. I found an answer in a Cygwin post - the `-i` parameter, so now I have my two 2.7 shells back, and I can continue slipping further into obscurity. Thanks again!

**Comment by gfixler** [comment]
> I'm assuming he himself cannot convert the code as I'm assuming Autodesk Maya only supports 2.7.11...

Correct. Many of us are desperate to switch to 3, as we're starting to run into problems like this. For now, though, even Maya 2021 is still on 2.7. Autodesk is afraid people won't upgrade if they're labyrinthine film/games pipelines might not Just Work in 3.x, both of which are probably true :(

Yes, I found 2.7 in `/mingw64/bin` (thanks!), but was getting a `LookupError: unknown encoding: cp65001` error. Long story short, my MSYS2's .bashrc now has `export PYTHONIOENCODING=UTF-8`, which solved that problem.

For many years I've had 2 pathways from Vim, from key mappings, one of which sends anything not tagged with "maya" (via nose's attr decorator) through 2.7 in the MSYS2 shell, and all tests, decoration or no, to Maya's mayapy, because the latter requires firing up some standalone stuff, which takes 5 or so seconds, and slows down my TDD testing loop, so if I'm just working on maya-agnostic bits, I can test instantly outside of Maya. You've made me realize that I could just send everything to mayapy, and just not fire up the slow, standalone stuff for the undecorated pathway.

Anyway, the second problem was that `/c/Program\ Files/Autodesk/Maya2018/bin/mayapy.exe` would sit there with no prompt, but I found an answer in a Cygwin post - the `-i` parameter - so now I have my two 2.7 shells back, and I can continue slipping further into obscurity.

Thanks again!
